src/validate.ts

```diff
@@ -39,33 +44,57 @@ function responseValidation(apiResponse: object, schema: object) {
  * @param apiResponse User defined api response.
  * @return Errors from AJV validation, if any. Undefined otherwise.
  */
-export function validate(intentRequest: object, apiResponse: object) {
+export function validate(intentRequest: object, apiResponse: object, syncData?: object){
   const responseType = intentRequest['inputs'][0]['intent'];
   if (responseType === 'action.devices.SYNC') {
-    // validate traits with attributes schema
+    const validateSyncAPI = responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA); 
     const syncErrors : object[] = [];
-    const syncDevices = apiResponse['payload']['devices'];
-    const syncDevicesLength = syncDevices.length;
-    for (let i = 0; i < syncDevicesLength; i++) {
-      const traits = syncDevices[i]['traits'];
-      const traitsLength = syncDevices[i]['traits'].length;
-      const attributes = syncDevices[i]['attributes'];
-      if (attributes === undefined) {
-        return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
-      }
-      for (let j = 0; j < traitsLength; j++) {
-        const trait = traits[j];
-        if (trait in TRAIT_ATTRIBUTES_EXPECT) {
-          const validateTraitRes = responseValidation(attributes, TRAIT_ATTRIBUTES_EXPECT[trait]);
-          if (validateTraitRes) {
-            return syncErrors.push(...validateTraitRes);
+    if (validateSyncAPI === undefined){
+        const syncDevices = apiResponse['payload']['devices'];
+        const syncDevicesLength = syncDevices.length;
+        for (let i = 0; i < syncDevicesLength; i++) {
+          const traits = syncDevices[i]['traits'];
+          const traitsLength = syncDevices[i]['traits'].length;
+          const attributes = syncDevices[i]['attributes'];
+          if (attributes === undefined) {
+            return validateSyncAPI;
           }
-        }
-      }
+          for (let j = 0; j < traitsLength; j++) {
+            const trait = traits[j];
+            if (trait in TRAIT_ATTRIBUTES_EXPECT) {
+              const validateTraitRes = responseValidation(attributes, TRAIT_ATTRIBUTES_EXPECT[trait]);
+              if (validateTraitRes) {
+                syncErrors.push(...validateTraitRes);
+              }
+            }
+          }
+       } 
+    } else {
+        syncErrors.push(...validateSyncAPI!);
     }
     return syncErrors.length ? syncErrors : undefined;
   } else if (responseType === 'action.devices.QUERY') {
-    // validate with states schema
+    // validate with states schema;
```

Comment:
Thanks for adding this into the PR. Looks like the validation order needs to be flipped to match the other cases.